cypress/e2e/trafficlight/trafficlight.spec.ts

```diff
@@ -121,23 +121,25 @@ function recurse() {
         const data: JSONValue = resp.body.data;
         const action: string = resp.body.action;
         cy.log('running action', action, JSON.stringify(data));
-        let result;
         try {
-            result = runAction(action, data);
+            cy.resolveFromPromise(runAction(action, data)).then(result => {
+                if (result) {
+                    sendResponse(result);
+                }
```

Comment:
Only certain actions like idle and exit return undefined. All other actions will return something (most likely a string).
If these other actions don't return something that means an error was thrown while they were being executed. We have hooked to the cypress events so that these are caught and reported to the server. See https://github.com/matrix-org/trafficlight-adapter-element-web/blob/51d7e790f133b40711f9d73e964d8d73c5a56dff/cypress/e2e/trafficlight/trafficlight.spec.ts#L56-L87

Reply:
I agree that this can probably be written in a better way but I suggest that we just create an issue for now (since this code was not introduced by the changes in this PR).